src/bootstrap/bootstrap.py

```diff
@@ -795,8 +785,13 @@ def get_toml(self, key, section=None):
         'true'
         """
 
+        if type(self) == str:
+            config_toml = self
+        else:
+            config_toml = self.config_toml
```

Comment:
I don't really understand how `self` can be something else than `RustBuild`, especially `str`.

Reply:
yeah, this was kinda hacky. if you look at the callsite, it's calling `RustBuild.get_toml` directly instead of `build.get_toml`: https://github.com/rust-lang/rust/blob/db8f6a095484a4226e65d12b52656f5529795d01/src/bootstrap/bootstrap.py#L1052 i did that to avoid having to add a free-standing get_toml function in addition to the class function, but i don't mind adding the free function if you prefer it.

Reply:
If this would be documented, it will also be fine.

Reply:
don't want to write documentation for this, i'll add the free function